src/common/threading.jl

```diff
@@ -1,42 +1,77 @@
 import FFTW
 using LinearAlgebra
 
-function setup_threading(; n_fft=1, n_blas=Threads.nthreads())
-    n_julia = Threads.nthreads()
+function setup_threading(; n_fft=1, n_blas=Threads.nthreads(), n_DFTK=nothing)
```

Comment:
Is `n_DFTK` a good name ? I think this is all potentially very confusing on the relationship between all these different threading parameters. Don't you think it would be clearer to just have a boolean flag to indicate whether we use the available Julia threads inside DFTK or not ?

My main grips here is that `n_DFTK` is a compile-time thing while all the other options here are runtime things. So using this function can easily lead to rather unexpected behaviour (one sometimes needs to exit the shell and recompile DFTK, sometimes not).

With a boolean flag and a separate function for that you avoid that issue.

Reply:
I think it makes sense to be able to set the number of threads that DFTK uses rather than just a flag. In some sense the new names are less confusing because they more accurately reflect what's going on: there are exactly three sources of threading, DFTK, BLAS and FFTW, and now we control the number of threads of all these three in this function.

The compile-time thing is confusing, but with the info printing it should be OK? I also added some safeguards.

Reply:
Add some docs to this function. Otherwise it's quite confusing now. (In particular mention that Julia threads also exist and are considered, so essentially n_DFTK and Julia threads configure the same thing: One per runtime and one per compile time.